### ISSUE TITLE
Loss vs. Accuracy In Course 1 - Part 4 - Lesson 2, Exercise 8

### ISSUE THREAD

**Issue by barnardb** [lmoroney/dlaicourse#9]
In `Course 1 - Part 4 - Lesson 2 - Notebook.ipynb`, Exercise 8 suggests it will use an accuracy cutoff, and then includes the following code:

```python
    if(logs.get('loss')<0.4):
      print("\nReached 60% accuracy so cancelling training!")
```

This seems to assume that loss and accuracy sum to 1, which is not the case. (Or am I very confused?)

To illustrate the problem, replace the second last line
```python
model.compile(optimizer='adam', loss='sparse_categorical_crossentropy')
```
wtih
```python
model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])
```
and rerun the code example.

For me, the accuracy is above 0.6 long before the loss drops below 0.4.

(There is some related confusion in the text just before the exercises, where following a result showing a loss of 0.3490932399511337 and accuracy of 0.8754 there is text reading "For me, that returned a loss of about .8838, which means it was about 88% accurate." I raised PR #7 to address this.)

**Comment by barnardb** [comment]
I just went on the next notebook in the course, `Course 1 - Part 4 - Lesson 4 - Notebook.ipynb`, and see that it correctly uses accuracy—and not loss—to implement its accuracy cutoff. Since using callbacks to implement cutoff is covered in the next lesson, perhaps Exercise 8 could simply be removed.

**Comment by barnardb** [comment]
And now I see that in the last video for this week, "Walk through a notebook with Callbacks", uses a loss cutoff, with a print statement in the code incorrectly claiming to be using accuracy, just like the code in Exercise 8.

**Comment by DarkArmed** [comment]
I just ran the Exercise code and got result as below.
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/resource_variable_ops.py:435: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
Epoch 1/10
60000/60000 [==============================] - 5s 78us/sample - loss: 0.4972 - acc: 0.8263
Epoch 2/10
60000/60000 [==============================] - 5s 76us/sample - loss: 0.3716 - acc: 0.8659
Epoch 3/10
60000/60000 [==============================] - 5s 77us/sample - loss: 0.3367 - acc: 0.8764
Epoch 4/10
60000/60000 [==============================] - 5s 77us/sample - loss: 0.3131 - acc: 0.8843
Epoch 5/10
60000/60000 [==============================] - 5s 91us/sample - loss: 0.2938 - acc: 0.8920
Epoch 6/10
60000/60000 [==============================] - 6s 92us/sample - loss: 0.2811 - acc: 0.8966
Epoch 7/10
60000/60000 [==============================] - 6s 94us/sample - loss: 0.2669 - acc: 0.9007
Epoch 8/10
60000/60000 [==============================] - 6s 93us/sample - loss: 0.2573 - acc: 0.9044
Epoch 9/10
60000/60000 [==============================] - 5s 78us/sample - loss: 0.2475 - acc: 0.9074
Epoch 10/10
60000/60000 [==============================] - 5s 79us/sample - loss: 0.2411 - acc: 0.9092
<tensorflow.python.keras.callbacks.History at 0x7f5679200e50>

I looked at official document （https://www.tensorflow.org/api_docs/python/tf/keras/losses）and found out that loss is the error between y_true and y_pred, but nothing to do with accuracy.

**Comment by chrismoroney** [comment]
@barnardb 
Check out this link here: https://datascience.stackexchange.com/questions/42599/what-is-the-relationship-between-the-accuracy-and-the-loss-in-deep-learning

This conversation clearly shows the definition of accuracy and loss.  While there is no relationship between the two of them, both metrics still contribute to your model's overall performance. 

In summary, accuracy determines how much of the data was classified correctly, while loss shows how much your model's predictions were off by (not as easily seen in binary classifiers, but is definitely more apparent in regression functions, and possibly multi-class classifiers). Below is a table that should help visualize these metrics.

| | Low loss | High Loss |
|------|------|------|
| Low Accuracy | Model's prediction was close but incorrect for much of the data, few predictions were correct | Model's prediction was not close for much of the data |
| High Accuracy | Model's prediction was correct for much of the data, few errors in some of the data  | Model's prediction was correct for much of the data, but what the Model got wrong was not close to correct answer |

In all, your callback function can be based both off of accuracy or loss. You are correct, however, in that the callback function stops based on the model's loss metric while it says "Reached 60% accuracy." It should either say prediction was less than 0.4 for loss or the input should be changed to accuracy.
